diff --git a/main.py b/main.py
index 11f3845..3a36c32 100644
--- a/main.py
+++ b/main.py
@@ -79,7 +79,7 @@ def parse_args():
 
 
 # GPU configuration use for faster processing
-device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+device = torch.device("cuda:1" if torch.cuda.is_available() else "cpu")
 
 
 # DNN modeling
@@ -175,9 +175,10 @@ class DQL:
 
             # Forward 
             # Loss calculation based on loss function
-            target_Q = target_Q.float()
-
-            Q_main = self.main_network(state).gather(1, action).squeeze()
+            Q_main = self.main_network(state).squeeze()
+            target_Q = target_Q.float() * torch.ones(torch.transpose(Q_main, 0, 1).shape, device=device)
+            target_Q = torch.transpose(target_Q, 0, 1)
+            
             loss = self.loss_func(target_Q.cpu().detach(), Q_main.cpu())
             # Intialization of the gradient to zero and computation of the gradient
             self.optimizer.zero_grad()
