diff --git a/contention_env_complex/__pycache__/uav_env.cpython-310.pyc b/contention_env_complex/__pycache__/uav_env.cpython-310.pyc
index 2bd72ae..0cb57f4 100644
Binary files a/contention_env_complex/__pycache__/uav_env.cpython-310.pyc and b/contention_env_complex/__pycache__/uav_env.cpython-310.pyc differ
diff --git a/contention_env_complex/main.py b/contention_env_complex/main.py
index c2c653b..373f340 100644
--- a/contention_env_complex/main.py
+++ b/contention_env_complex/main.py
@@ -74,7 +74,7 @@ def parse_args():
     parser.add_argument("--uav-dis-th", type=int, default=1000, help="distance value that defines which uav agent share info")
     parser.add_argument("--dist-pri-param", type=float, default=1/5, help="distance penalty priority parameter used in level 3 info exchange")
     parser.add_argument("--coverage-threshold", type=int, default=75, help="if coverage threshold not satisfied, penalize reward, in percentage")
-    parser.add_argument("--coverage-penalty", type=int, default=5, help="penalty value if threshold is not satisfied")
+    parser.add_argument("--coverage-penalty", type=int, default=10, help="penalty value if threshold is not satisfied")
 
 
     args = parser.parse_args()
