diff --git a/dynamic_env/main.py b/dynamic_env/main.py
index cb7cd51..0328bec 100644
--- a/dynamic_env/main.py
+++ b/dynamic_env/main.py
@@ -206,6 +206,7 @@ if __name__ == "__main__":
             name = run_id,
             save_code= True,
         )
+
     # Track everyruns inside run folder // Tensorboard files to keep track of the results
     writer = SummaryWriter(f"runs/{run_id}")
     # Store the hyper paramters used in run as a Scaler text inside the tensor board summary
@@ -329,7 +330,7 @@ if __name__ == "__main__":
         writer.add_scalar("charts/episodic_length", num_epochs, i_episode)
         writer.add_scalar("charts/connected_users", episode_user_connected[i_episode], i_episode)
         if args.wandb_track:
-            wandb.log({"episodic_reward": episode_reward[i_episode], "episodic_length": num_epochs, "connected_users":episode_user_connected[i_episode] })
+            wandb.log({"episodic_reward": episode_reward[i_episode], "episodic_length": num_epochs, "connected_users":episode_user_connected[i_episode]})
             wandb.log({"reward: "+ str(agent): reward[agent] for agent in range(NUM_UAV)})
             wandb.log({"connected_users: "+ str(agent_l): user_connected[agent_l] for agent_l in range(NUM_UAV)})
 
@@ -391,8 +392,8 @@ if __name__ == "__main__":
             #############################
             # writer.add_figure("images/uav_users", figure, i_episode)
 
-            writer.add_scalar("chart/connected_users_after", sum(temp_user_2))
-            writer.add_scalar("chart/connected_users_before", sum(temp_user_1))
+            writer.add_scalar("charts/connected_users_after", sum(temp_user_2), i_episode)
+            writer.add_scalar("charts/connected_users_before", sum(temp_user_1), i_episode)
 
             print(drone_act_list)
             print("Number of user connected before change in ",i_episode," episode is: ", temp_user_1)
@@ -469,5 +470,6 @@ if __name__ == "__main__":
     writer.add_text("best outcome_before", str(best_state_1))
     writer.add_text("best outcome_after", str(best_state_2))
 
+    wandb.finish()
     writer.close()
 
diff --git a/dynamic_env/run.py b/dynamic_env/run.py
index e8a533d..3fc83f8 100644
--- a/dynamic_env/run.py
+++ b/dynamic_env/run.py
@@ -6,4 +6,4 @@ for i in range(1, 5):
     print('####  Running the code for Level:', i, "info exchange  ####")
     print('#######################################################')
     p = subprocess.run(["python", "uav_env.py"])
-    g = subprocess.run(["python", "main.py", "--info-exchange-lvl", str(i), "--wandb-track", "True"])
\ No newline at end of file
+    g = subprocess.run(["python", "main.py", "--info-exchange-lvl", str(i), "--wandb-track", "True", "--num-episode", str(800)])
\ No newline at end of file
