diff --git a/run.py b/run.py
index a8362c6..b34d5c6 100644
--- a/run.py
+++ b/run.py
@@ -6,7 +6,7 @@ for i in range(1, 5):
     print('####  Running the code for Level:', i, "info exchange  ####")
     print('#######################################################')
     p = subprocess.run(["python", "uav_env.py"])
-    g = subprocess.run(["python", "main.py", "--info-exchange-lvl", str(i), "--num-episode", str(450), "--wandb-track", "True", "--learning-rate", str(1e-4)])
+    g = subprocess.run(["python", "main.py", "--info-exchange-lvl", str(i), "--num-episode", str(450), "--wandb-track", "True", "--learning-rate", str(3.5e-4)])
 
 # ## Running for level 3 with different distance values
 for i in range(0,1000, 250):
@@ -14,4 +14,4 @@ for i in range(0,1000, 250):
     print('####  Running the code for distance threshold:', i, " with Level: 2 ####")
     print('###########################################################')
     p = subprocess.run(["python", "uav_env.py"])
-    g = subprocess.run(["python", "main.py", "--info-exchange-lvl", str(2), "--num-episode", str(800), "--wandb-track", "True", "--uav-dis-th", str(i), "--learning-rate", str(1e-4)])
+    g = subprocess.run(["python", "main.py", "--info-exchange-lvl", str(2), "--num-episode", str(800), "--wandb-track", "True", "--uav-dis-th", str(i), "--learning-rate", str(3.5e-4)])
